### DISCUSSION TITLE
[ How To ] how to deploy this to Cloudflare Pages?

### DISCUSSION THREAD

**Discussion by alerezaaa** [alshedivat/al-folio#775] [Q&A] [answered]
Hi

I tried to deploy this to Cloudflare Pages but I couldnt

CF-pages doesnt support Ruby Version more 2.7.6 but `al-folio` need `3.0.2`

So

how can I deploy it to cloudflare pages

**Comment by rohandebsarkar** [comment]
The easiest way, will be to use the default GitHub Actions to generate the site and push it to the `gh-pages` branch. Then you can use the Cloudflare pages to deploy the `gh-pages` branch. You may disable GitHub Pages if you don't want it.

You may also use the `bin/deploy` script to push the generated site to another branch, for that read the `README.md`. But this may need some advanced knowledge.

**Comment by alerezaaa** [comment]
thanks for your response
I did
I create a `cf-pages` and use it as `root directory` in cloudflare pages settings but, the `[ci skip]` phrase at the end of commit which causes cloudflare to stop auto deployement (so, next updates wont be deployed)
unfortunatly, I dont know what this `bin/deploy` is, to figure it out how can I solve this


a part of github action log:
```
+ touch .nojekyll
+ git add -fA
++ git log -1 --pretty=%B
+ git commit --allow-empty -m 'Update deploy.yml [ci skip]'
[cf-pages 6220081] Update deploy.yml [ci skip]
 198 files changed, 448 insertions(+), 27069 deletions(-)
 delete mode 100644 .all-contributorsrc
```

**Comment by rohandebsarkar** [comment, marked as answer]
You can remove the `[ci-skip]` by changing the following line in the `bin/deploy` script.

https://github.com/alshedivat/al-folio/blob/99b14c436ffe674a9670b901aeebdc4897625de3/bin/deploy#L110

Also, to make GitHub Actions push the generated site on any branch other than `gh-pages`, you will need to edit the following line in the `.github/workflows/deploy.yml` script.

https://github.com/alshedivat/al-folio/blob/99b14c436ffe674a9670b901aeebdc4897625de3/.github/workflows/deploy.yml#L38

You can change the `gh-pages` in the above line to `cf-pages` that will make GitHub Actions automatically push to the `cf-pages` branch.

**Comment by rohandebsarkar** [comment]
Converting this issue into a discussion.

**Comment by pourmand1376** [comment]
Currently. I host my website on `Netlify` and then use `Cloudflare` as CDN!
